test/sanity/validate-modules/ignore.txt

```diff
@@ -1532,6 +1532,7 @@ lib/ansible/modules/remote_management/ipmi/ipmi_boot.py E326
 lib/ansible/modules/remote_management/ipmi/ipmi_power.py E326
 lib/ansible/modules/remote_management/manageiq/manageiq_alert_profiles.py E324
 lib/ansible/modules/remote_management/manageiq/manageiq_alerts.py E324
+lib/ansible/modules/remote_management/manageiq/manageiq_dynamic_resource_definition.py E324
```

Comment:
New ignores should not be added. The reported issues should be fixed. If the issues exist in common docs fragments then those should be fixed in a separate PR first.

Reply:
@mattclay see [above](https://github.com/ansible/ansible/pull/38725#issuecomment-381643640) why I believe linter is wrong here.  Common doc fragment & argspec agree on default here.

`test/sanity/validate-modules/main.py` does understand shared doc fragments.
But it gets confused by suboptions that have a default:
https://gist.github.com/cben/14de325ceec502d86c1edbcd082be9c2
There is `TODO: needs to recursively traverse suboptions` :-).  I'm not familiar with it enough really implement that, but I'm working on patch to not produce spurious error in such scenario.

Reply:
This error has brought to my attention a bug in the options parser.  The use of `default` on an option using `options`  with `default` on the suboptions, is a work around for the real problem.

A PR with the resolution can be found at: https://github.com/ansible/ansible/pull/38967

Once this is resolved, `default=` can be removed from the `manageiq_connection` option.

Reply:
Great find!  I forgot all about the workround in manageiq modules (added in #31774) despite reviewing it :)